shell/platform/fuchsia/flutter/software_surface.cc

```diff
@@ -0,0 +1,404 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "software_surface.h"
+
+#include <lib/async/default.h>
+#include <lib/ui/scenic/cpp/commands.h>
+#include <zircon/status.h>
+#include <zircon/types.h>
+
+#include <algorithm>
+
+#include "flutter/fml/logging.h"
+#include "flutter/fml/trace_event.h"
+#include "include/core/SkImageInfo.h"
+#include "third_party/skia/include/core/SkColorSpace.h"
+#include "third_party/skia/include/core/SkImageInfo.h"
+
+#include "../runtime/dart/utils/inlines.h"
+
+namespace flutter_runner {
+
+namespace {
+
+constexpr SkColorType kSkiaColorType = kRGBA_8888_SkColorType;
+
+}  // namespace
+
+SoftwareSurface::SoftwareSurface(
+    fuchsia::sysmem::AllocatorSyncPtr& sysmem_allocator,
+    fuchsia::ui::composition::AllocatorPtr& flatland_allocator,
+    scenic::Session* session,
+    const SkISize& size,
+    uint32_t buffer_id)
+    : session_(session), wait_(this) {
+  FML_CHECK(session_ || flatland_allocator.is_bound());
```

Comment:
Should we do `  FML_CHECK((session_ == nullptr) ^ flatland_allocator.is_bound());` instead?

Having both seems like an invalid state.

Reply:
Done